cpp/dolfinx/fem/discreteoperators.h

```diff
@@ -74,22 +77,21 @@ void discrete_gradient(const FunctionSpace& V0, const FunctionSpace& V1,
     throw std::runtime_error("Block size is greather than 1 for V1.");
 
   // Get V0 (H(curl)) space interpolation points
-  const xt::xtensor<double, 2> X = e1->interpolation_points();
+  const auto [X, Xshape] = e1->interpolation_points();
 
   // Tabulate first order derivatives of Lagrange space at H(curl)
   // interpolation points
   const int ndofs0 = e0->space_dimension();
   const int tdim = mesh->topology().dim();
-  xt::xtensor<double, 4> phi0
-      = xt::empty<double>({tdim + 1, int(X.shape(0)), ndofs0, 1});
-  e0->tabulate(phi0, X, 1);
+  std::vector<double> phi0_b((tdim + 1) * Xshape[0] * ndofs0 * 1);
```

Comment:
```suggestion
  std::vector<double> phi0_b((tdim + 1) * Xshape[0] * ndofs0 * 1, 0);
```

Reply:
I don't see why zeroing is required (it's implicit anyway).